kyo-prelude/shared/src/main/scala/kyo/Aspect.scala

```diff
@@ -4,180 +4,223 @@ import Aspect.*
             local.let(map + (this -> cut))(v)
         }
 end Aspect
 
-/** Companion object for Aspect class. */
 object Aspect:
 
     private[kyo] val local = Local.init(Map.empty[Aspect[?, ?, ?], Cut[?, ?, ?]])
 
-    /** Represents a cut in the Kyo aspect-oriented programming model.
+    /** Represents a cut in the aspect-oriented programming model.
       *
-      * A cut defines how an aspect modifies or intercepts a computation. It provides a way to inject behavior before, after, or around a
-      * given computation.
+      * A cut defines how an aspect modifies or intercepts computations. It provides a way to inject behavior before, after, or around the
+      * intercepted operations.
       *
-      * @tparam A
-      *   The input type of the cut
-      * @tparam B
-      *   The output type of the cut
+      * @tparam Input
+      *   The input type constructor
+      * @tparam Output
+      *   The output type constructor
       * @tparam S
-      *   The effect type of the cut
+      *   The effect type
       */
-    abstract class Cut[A, B, S]:
-
-        /** Applies this cut to the given computation.
+    abstract class Cut[Input[_], Output[_], S]:
+        /** Applies this cut to transform a computation.
           *
-          * @param v
-          *   The input value
-          * @param f
-          *   The function to apply after the cut
-          * @tparam S2
-          *   Additional effects
-          * @return
-          *   The result of applying the cut and then the function
+          * @param input
+          *   The input value to transform
+          * @param cont
+          *   The continuation function to apply after transformation
+          * @tparam C
+          *   The type parameter for Input/Output
           */
-        def apply(v: A)(f: A => B < S): B < S
+        def apply[C](input: Input[C])(cont: Input[C] => Output[C] < S): Output[C] < S
 
-        /** Chains this cut with another cut.
+        /** Chains this cut with another, creating a new cut that applies both transformations in sequence.
           *
           * @param other
-          *   The cut to chain after this one
-          * @return
-          *   A new cut that represents the chained cuts
+          *   The cut to apply after this one
+          */
+        def andThen(other: Cut[Input, Output, S])(using Frame): Cut[Input, Output, S] =
+            new Cut[Input, Output, S]:
+                def apply[A](input: Input[A])(cont: Input[A] => Output[A] < S): Output[A] < S =
+                    Cut.this(input)(other(_)(cont))
+    end Cut
+
+    object Cut:
+
+        /** Represents a binding function that implements a cut.
+          *
+          * @tparam I
+          *   The input type constructor
+          * @tparam O
+          *   The output type constructor
+          * @tparam S
+          *   The effect type
+          */
+        type Binding[I[_], O[_], S] = [C] => (I[C], I[C] => O[C] < S) => O[C] < S
```

Comment:
I've added this new `Cut.Binding` generic function to make it easier to create cuts with anonymous functions.

Reply:
I've decided to remove `Binding` and define `Cut` as the generic function directly to simplify the effect.